### ISSUE TITLE
Add IDIR authentication protection to API endpoints

### ISSUE THREAD

**Issue by andrea-williams** [bcgov/cas-registration#211]
#### Describe the task

We need to implement an authentication mechanism to all our API endpoints to ensure that only logged-in users (whether with IDIR or Business BCeID) can successfully send requests to our API.

Any not-authenticated user should receive a 401 Unauthorized response when they send a request to our API.

Additionally, some API endpoints should only be available to internal staff (IDIR users).

This means that each endpoint will require a token to be included in the request.

#### Additional context

~~- in our SSO integration, we've already got an "admin" role for our Dev environment - this is the intended role for CAS internal staff. (Maybe this should be renamed to something like `cas_admin`?)~~
- we have 3 app_roles relevant to IDIR users: `cas_admin`, `cas_analyst`, and `cas_pending`

#### Acceptance Criteria

- [ ] a request with no token or an invalid token receives a 401 Unauthorized response status code when the request is sent to any of our API endpoints

Our API endpoints:
- [x] GET /naics_codes: cas_admin & cas_analyst permitted
~~- [ ] GET /naics_categories: cas_admin permitted~~ (/naics_categories endpoint deleted)
- [x] GET /operations: cas_admin & cas_analyst permitted
- [x] POST /operations: cas_admin/cas_analyst NOT permitted (have confirmed with business area; there will never be a need for internal staff to create a new operation on behalf of an IO)
- [ ] GET /operations/{operation_id}: cas_admin & cas_analyst permitted
- [ ] PUT /operations/{operation_id}: cas_admin & cas_analyst permitted
- [x] PUT /operations/{operation_id}/update-status: cas_admin & cas_analyst permitted
- [x] GET /operators: cas_admin & cas_analyst permitted
- [ ] GET /operators/{operator_id}: cas_admin & cas_analyst permitted
- [ ] GET /select-operator/{operator_id}: cas_admin/cas_analyst NOT permitted
- [ ] GET and PUT /select-operator/user-operator/{user_operator_id}: cas_admin & cas_analyst permitted (I think this will be necessary for approving prime admin. Implementation pending)
- [ ] for now, assume that cas_pending doesn't have permission for anything

- [ ] update backend unit tests to include test cases to check that authentication works as expected

#### Additional additional context
- A separate ticket #216 has been created for Business BCeID users
- I'm not entirely sure whether cas_admin will need permissions for PUT /operations/{operation_id} - we'll add it for now to be on the safe side...

**Comment by Dianadec** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Team-Moose-Sprint-Board-60d216175a2689000ec774d0/issues/bcgov/cas-registration/211?planning-poker) with Zenhub @andrea-williams @joshgamache @JoshLarouche @shon-button @dleard @mikevespi @Sepehr-Sobhani @BCerki

**Comment by BCerki** [comment]
Maybe we could split this card into two cards, one for IDIR-related endpoints and one for BCEID. Otherwise this card is blocked by both IDIR and BCEID setup, and it's a big card.

**Comment by Dianadec** [comment]
Once we have official list of permissions and roles, let's document them.
